site-manage/site-manage-group-section-role-helper/tool/src/java/org/sakaiproject/site/tool/helper/managegroupsectionrole/impl/SiteManageGroupSectionRoleHandler.java

```diff
@@ -1499,14 +1500,15 @@ public int compare(Object o1, Object o2) {
      * @return status
      */
     public String processUploadAndCheck() {
-        String uploadsDone = (String) httpServletRequest.getAttribute(RequestFilter.ATTR_UPLOADS_DONE);
+        HttpServletRequest request = (HttpServletRequest) ComponentManager.get(ThreadLocalManager.class).get(RequestFilter.CURRENT_HTTP_REQUEST);
```

Comment:
I think it's worth commenting why you can't use the standard injected `HttpServletRequest` that RSF provides here.

Reply:
We don’t seem to be able to inject the current HttpServletRequest because there isn’t a factory that I can see which gets the current request, just a factory that returns the request when the factory was created. We need a better factory as this bit of the code is sharing a request scope bean across multiple requests (bad). 

This hack just pulls the request out of the thread local which should always work in the Sakai world but it’s not a nice fix.

Thanks Matthew :-)